lib/Dialect/ESI/runtime/cpp/lib/Accelerator.cpp

```diff
@@ -19,6 +19,13 @@
 #include <stdexcept>
 
 namespace esi {
+services::Service *Accelerator::getServiceImpl(Service::Type svcType) {
+  std::unique_ptr<Service> &cacheEntry = serviceCache[&svcType];
+  if (cacheEntry != nullptr)
+    return cacheEntry.get();
+  cacheEntry = std::unique_ptr<Service>(createService(svcType));
+  return cacheEntry.get();
+}
```

Comment:
use `std::map::emplace` instead.

Reply:
Getting a reference to the storage slot saves a little code (which I fucked up) and a search for the insertion.